logicalRequirements.md

```diff
@@ -242,6 +242,22 @@ __Example:__
 }}
 ```
 
+#### lavaFramesWithEnergyDrops object
+
+A `lavaFramesWithEnergyDrops` object represents the need for Samus to spend time in lava, but with the possibility of offsetting some of the lava damage using energy drops from enemies. Any lava damage is logically applied before the energy gain, so Samus must be able to survive the heat before picking up the drops. Any energy gain is logically capped to not exceed the heat damage, so this logical requirement cannot result in a net energy gain.
```

Comment:
> Heat damage

Every room except this one has heat+lava damage simultaneously.  In SMZ3, Gravity grants full protection from lava and Varia full protection from heat.  So I think the usage would be running both lavaFramesWithEnergyDrops and HeatFrameWithEnergyDrops in parallel so the randomizer picks which is better for the situation.